### PR TITLE
feat(push): notify the auth-server when profile updated

### PR DESCRIPTION
**Pull request by eoger**
Connects to https://github.com/mozilla/fxa-features/issues/118

### CODE REVIEW

File: lib/routes/avatar/upload.js [hunk 1]
```diff
@@ -55,6 +59,7 @@ module.exports = {
         return db.addAvatar(id, uid, url, FXA_PROVIDER);
       })
       .done(function uploadDone() {
+        profileUpdatedQueue.send({ uid: uid.toString('hex') });
```
Comment (seanmonstar) [inline lib/routes/avatar/upload.js]: It looks like this returns a `Promise`, over in the `sqs.js` file. Should we wait on it before replying, or is ignoring it on purpose? If it is, may be worth a comment acknowledging that so Future Us can remember.

Comment (eoger) [inline lib/routes/avatar/upload.js]: This is on purpose, will add a comment.

File: lib/config.js [hunk 2]
```diff
@@ -221,6 +221,20 @@ const conf = convict({
...
+      env: 'AUTH_SERVER_MESSAGING_REGION',
+      default: ''
+    },
+    profileUpdatesQueueUrl: {
+      doc: 'The bounce queue URL to use (should include https://sqs.<region>.amazonaws.com/<account-id>/<queue-name>)',
+      format: String,
+      env: 'PROFILE_UPDATES_QUEUE_URL',
+      default: ''
+    }
+  },
```
Comment (rfk) [inline lib/config.js]: Let's get @jbuck and @jrgm input on this as well; guys, thoughts on using an SQS queue here to relay notifications back to the auth-server, and from there on to the connected devices?

File: lib/routes/avatar/upload.js [hunk 3]
```diff
@@ -7,16 +7,20 @@ const assert = require('assert');
...
+const SQSSender = require('../../sqs')(logger);
 const validate = require('../../validate');
 const workers = require('../../img-workers');
 
 const FXA_PROVIDER = 'fxa';
 const FXA_URL_TEMPLATE = config.get('img.url');
 assert(FXA_URL_TEMPLATE.indexOf('{id}') !== -1,
     'img.url must contain the string "{id}"');
+const profileUpdatedQueue = new SQSSender(config.get('authServerMessaging.region'),
+                                          config.get('authServerMessaging.profileUpdatesQueueUrl'));
```
Comment (rfk) [inline lib/routes/avatar/upload.js]: I wonder if we could push this down a level of abstraction, so we can create this `SQSSender` object in one place, and then all these other modules just have to import something and pass it the target uid.

**Comment by eoger** [comment]
Thanks Ryan I put the queue creation in another module.

File: lib/config.js (hunk 2 above)
Comment (seanmonstar) [inline lib/config.js]: I don't recall, did we get an answer? I want to say it was "yea, sure, do it", but fuzzy.

Comment (eoger) [inline lib/config.js]: IIRC, @rfk told me they discussed this in a meeting and the idea was accepted.

Comment (rfk) [inline lib/config.js]: Yea, sure, do it. 👍

**Pull request merged**